docs(core): add a page for Nx releases (#23455)
nrwl/nx: docs/shared/reference/releases.md

# Releases

Major Nx versions are released as the _latest_ every six months, typically around April and October.
After each major version release, the _previous_ major version moves to long-term support (LTS) for 12 months, after
which it becomes unsupported.
Each major version has 18 total months of support from when it is first released to when it falls out of LTS.

> Workspaces should only use major versions that are under LTS.

Reviewer: Mm... this sentence is weird to me. Obviously, people should be on the latest version? :sweat_smile:  It's also weird because the current version is not LTS yet?

In the same vein, we can encourage people to keep up to date with Nx by saying something along the lines of "Use `nx migrate` often to stay up to date with Nx.

> ## Supported Versions

Reviewer: I think we should include a bit on the difference between Major, Minor, and Patch releases.

Mostly highlighting that most breaking changes are reserved for major versions. Minor versions new features and patch versions include bug fixes. Updating to the latest minor or patch version should be safe to do regularly.

The following are the currently supported versions of Nx.

> | Version | Release Date |

Reviewer: I think we should include the type of support it's in. (Current vs LTS)

| ------- | ------------ |
| v19     | 2024-05-06   |
| v18\*   | 2024-02-03   |
| v17     | 2023-10-19   |
| v16     | 2023-04-27   |

**\*Note:** v18 is a special release and does not fit into the normal release cycle. Thus, v16 continues to be supported
according to schedule.

### LTS Fixes

> LTS fixes are generally considered if they address a security vulnerability or a critical bug that impedes the use of

Reviewer: This is worded weirdly as well. Maybe something more along the lines of...

Current vs LTS

The current version of Nx will receive new features as well as any fixes to any unintentional behavior. When a new major version of Nx is released, previous versions will go into Long Term Support (LTS). LTS versions of Nx will receive security patches as well as critical fixes.

that version of Nx.

## Deprecation Policy

When the Nx team intends to remove an API or feature, it will be marked as _deprecated_. The deprecated API will remain

> functional for two major versions (or one year), after which they will be removed.

Reviewer: ```suggestion
functional for a whole major version, after which they will be removed.
```

> 

Reviewer: I think an explicit example may help here. Something along the lines of... If something is deprecated in `16.2.0`, it will still work as expected for the entirety of Nx 17. When Nx 18 is released, the feature would be removed.

Also, add more description stating that... these deprecations may show up in the documentation, as warnings in console output, and/or TS Doc.
